pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/TableUpsertMetadataManagerFactory.java

```diff
@@ -83,8 +82,14 @@ public static TableUpsertMetadataManager create(TableConfig tableConfig,
                 metadataManagerClass, tableNameWithType), e);
       }
     } else {
-      LOGGER.info("Creating ConcurrentMapTableUpsertMetadataManager for table: {}", tableNameWithType);
-      metadataManager = new ConcurrentMapTableUpsertMetadataManager();
+      if (upsertConfig.isEnableConsistentDeletes()) {
```

Comment:
hmm.. if metadataManagerClass is set, the else-branch is skipped, so the check of the new feature flag is skipped too.

so should we check the feature flag check inside ConcurrentMapTableUpsertMetadataManager getOrCreatePartitionManager(), and based on the flag, we return either the original partition mgr, or the new partition mgr added in this PR.

Reply:
Since the new `ConcurrentMapTableUpsertMetadataManagerForConsistentDeletes` is almost the same as `ConcurrentMapTableUpsertMetadataManager`, I guess we can move the logic into `ConcurrentMapTableUpsertMetadataManager.getOrCreatePartitionManager()` and simply change the value type of `_partitionMetadataManagerMap` to `BasePartitionUpsertMetadataManager`

Reply:
Yes thanks Jackie for this suggestion! Looks neat now.